src/sentry/snuba/metrics/query_builder.py

```diff
@@ -590,26 +599,36 @@ def translate_results(self):
 
         # Applying post query operations for totals and series
         for group in groups:
-            totals, series = group["totals"], group["series"]
+            totals = group.get("totals")
+            series = group.get("series")
+
             for op, metric_name in self._bottom_up_dependency_tree:
                 metric_obj = metric_object_factory(op=op, metric_name=metric_name)
-                # Totals
-                totals[metric_name] = metric_obj.run_post_query_function(totals)
-                # Series
-                for idx in range(0, len(self._intervals)):
-                    series.setdefault(
-                        metric_name,
-                        [metric_obj.generate_default_null_values()] * len(self._intervals),
+                if totals is not None:
+                    totals[metric_name] = metric_obj.run_post_query_function(
+                        totals, query_definition=self._query_definition
                     )
-                    series[metric_name][idx] = metric_obj.run_post_query_function(series, idx)
+
+                if series is not None:
+                    # Series
+                    for idx in range(0, len(self._intervals)):
+                        series.setdefault(
+                            metric_name,
+                            [metric_obj.generate_default_null_values()] * len(self._intervals),
+                        )
+                        series[metric_name][idx] = metric_obj.run_post_query_function(
+                            series, query_definition=self._query_definition, idx=idx
+                        )
 
         # Remove the extra fields added due to the constituent metrics that were added
         # from the generated dependency tree. These metrics that are to be removed were added to
         # be able to generate fields that require further processing post query, but are not
         # required nor expected in the response
         for group in groups:
-            totals, series = group["totals"], group["series"]
-            for key in copy.deepcopy(list(totals.keys())):
+            totals = group.get("totals")
+            series = group.get("series")
+
+            for key in set(totals or ()) | set(series or ()):
```

Comment:
I don't understand this loop condition.. What does this do?

Reply:
I thought I responded to this... basically since `totals` can be `None`, we can't use it to iterate over existing fields. this takes both dicts, and unites their keys into a single set. `set(None)` is invalid so I use `set(totals or ())` to create an empty set if `totals` is `None`.